src/main/scala/com.snowplowanalytics.sauna/observers/S3Observer.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2015 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.sauna.observers
+
+import java.io.InputStream
+import java.net.URLDecoder._
+
+import awscala.s3.{Bucket, S3}
+import awscala.sqs.{Queue, SQS}
+import com.snowplowanalytics.sauna.loggers.Logger
+import com.snowplowanalytics.sauna.processors.Processor
+import play.api.libs.json.Json
+
+/**
+  * Observes some AWS S3 bucket.
+  */
+class S3Observer(s3: S3, sqs: SQS, queue: Queue,
```

Comment:
Probably I didn't understand you comment. This class extends `Observer`, and values `s3: S3, sqs: SQS, queue: Queue` are problem-specific and unlikely to happen elsewhere.